dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/helpers/calculate-initial-utilisation-and-fixed-fee.ts

```diff
@@ -0,0 +1,82 @@
+import { calculateInitialUtilisation, isValidDate } from '@ukef/dtfs2-common';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { calculateInitialFixedFee } from './calculate-initial-fixed-fee';
+
+export type RequiredParams = {
+  value?: number | null;
+  interestPercentage?: number | null;
+  dayCountBasis?: number | null;
+  coverStartDate?: string | Date | null;
+  coverEndDate?: string | Date | null;
```

Comment:
Do we need to do `number | null;` and `string | Date | null;`, since they're optional - would the following work? E.g

```
value?: number;
coverStartDate?: string | Date;
```

Reply:
doesn't work as it gets `string | Date | null` from the facility and ? is undefined so causes a linting error

Reply:
Would be nice to refactor so that it returns something more consistent rather than 3x different optional types; One for another day...